packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl

```diff
@@ -4,16 +4,25 @@
 // To add platforms, run `flutter create -t plugin --platforms <platforms> .` under the same
 // directory. You can also find a detailed instruction on how to add platforms in the `pubspec.yaml` at https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin-platforms.
 {{/no_platforms}}
-
 import 'dart:async';
 
-import 'package:flutter/services.dart';
+import '{{projectName}}_platform_interface.dart';
 
 class {{pluginDartClass}} {
-  static const MethodChannel _channel = MethodChannel('{{projectName}}');
+  factory {{pluginDartClass}}() {
+    _singleton ??= {{pluginDartClass}}._();
+    return _singleton!;
+  }
+
+  {{pluginDartClass}}._();
+
+  static {{pluginDartClass}}? _singleton;
+
+  static {{pluginDartClass}}Platform get _platform {
+    return {{pluginDartClass}}Platform.instance;
+  }
 
-  static Future<String?> get platformVersion async {
-    final String? version = await _channel.invokeMethod('getPlatformVersion');
-    return version;
+  Future<String?> get platformVersion {
```

Comment:
Let's change this from `get platformVersion` to `getPlatformVersion()`. Using a getter here is [against Flutter style](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods) (not that we have to follow Flutter style in a package, but the arguments that are made there are generally applicable), and even more importantly it doesn't scale to anything with arguments, making it harder to follow as a modifiable starting point.